pan3d/dataset_builder.py

```diff
@@ -171,7 +231,34 @@ def set_dataset_path(self, dataset_path, **kwargs):
     def on_set_array_active(self, array_active, **kwargs):
         if array_active is None or not self.state.dataset_ready:
             return
-        self.state.coordinates = list(self.data_array.coords.keys())
+        da = self.data_array
+        for key in da.coords.keys():
+            array_min = float(da.coords[key].min())
+            array_max = float(da.coords[key].max())
+            coord_attrs = [
+                {"key": k, "value": v} for k, v in da.coords[key].attrs.items()
+            ]
+            coord_attrs.append({"key": "dtype", "value": str(da.coords[key].dtype)})
+            coord_attrs.append({"key": "length", "value": da.coords[key].size})
+            coord_attrs.append(
+                {
+                    "key": "range",
+                    "value": [array_min, array_max],
+                }
+            )
+            self.state.coordinates.append(
+                {
+                    "name": key,
+                    "attrs": coord_attrs,
+                    "size": da.coords[key].size,
+                    "range": [array_min, array_max],
+                    "start": array_min,
+                    "stop": array_max,
+                    "step": 1,
+                }
+            )
+            self.state.expanded_coordinates.append(key),
+        self.auto_select_coordinates()
```

Comment:
Since you are editing in place `expanded_coordinates` and `coordinates`. Don't you need to mark them dirty?

Reply:
I hadn't noticed a case in which they don't update, but I'll add that line in anyway to protect against it.

Reply:
Probably something updates the content or marks them dirty somewhere else. But it is good practice to be explicit if you aim that variable to be properly shared with the client.

Reply:
Added in [910fe0c](https://github.com/Kitware/pan3d/pull/29/commits/910fe0c13c7e97599837f028138bcb959a66f4dc)